### ISSUE TITLE
Basic usage example only logs the messages half the time

### ISSUE THREAD

**Issue by stackotter** [sushichop/Puppy#74]
With Xcode 14.1 I have found that running the basic usage example from the readme is quite temperamental. Either both the messages are logged or nothing is logged. Adding a `fatalError` or `print("hi")` or `sleep(10)` to the end of the example all seem to fix the issue. This leads me to believe that it is something to do with batching of logs not working correctly when the last bit of code to run is a line that logs a message.

I have provided the simple Swift package that reproduces this issue. To reproduce the issue just run `swift run` multiple times and eventually you should get lucky and no log messages will be printed. Let me know if you need any additional context or you can't reproduce the issue yourself.

[LogTest.zip](https://github.com/sushichop/Puppy/files/10430241/LogTest.zip)

**Comment by sushichop** [comment]
@stackotter 
Thanks for your opening issue.
I have checked your sample code. This is because each logger is running asynchronously on each queue(not main queue) internally.

As a new feature, Puppy may need to be added a callback method to receive the completion of logging execution.

Thanks.

**Comment by stackotter** [comment]
All good, there's absolutely no rush on my end as this isn't really a huge issue for my application (it only happens in very rare situations), I just thought you should know :)

**Comment by sushichop** [comment]
Thanks for letting me know🙂

**Comment by stackotter** [comment]
Just let me know if there's something I can PR to help fix it, I'd be happy to help fix it :) I just don't know much about the internals of the library so it's tricky for me to find the cause

**Comment by doozMen** [comment]
This is indeed also the issue I discovered and tried to workaround in #76 but I made it too complex and rely on swift async/await features that are unavailable at the moment.

@sushichop please let me know if I can help in any way to add this. I feel maybe just using `DispatchGroup` on every `Lobberable` might work?

Side note: Discussion on swift forum https://forums.swift.org/t/swift-log-api-add-async-await-to-avoid-problems-in-async-code/62913/5

**Comment by sushichop** [comment]
@doozMen 
Sure. I'll do it🙂
Thanks for your comment.

**Comment by sushichop** [comment]
It may not so smart, but I have added a method to flush log message.
https://github.com/sushichop/Puppy/pull/79/files

```swift
log.info("INFO message for testFlush")
log.notice("NOTICE message for testFlush")
_ = log.flush()
```

The default value of timeout is 3 seconds. But since `DispatchGroup` is used internally, it will actually return as soon as the logging processing has finished in most cases.

In the future, I may implement a similar feature in smarter way.

Thanks.

**Comment by sushichop** [comment]
I have released [Puppy 0.7.0](https://github.com/sushichop/Puppy/releases/tag/0.7.0).
It includes the above fix, so I'm closing this issue.

Please feel free to reopen it if you have any questions.

Thanks.
